device/core/src/device_client.ts

```diff
@@ -171,13 +176,85 @@ export class Client extends InternalClient {
+        if (!err) {
+          debug('got blob storage shared access signature.');
+          /*Codes_SRS_NODE_DEVICE_CLIENT_41_004: [The `getBlobSharedAccessSignature` method shall store the `result` value for use by the `notifyBlobUploadStatus` method.]*/
+          this._blobStorageUploadParams = result;
+        } else {
+          debug('Could not obtain blob shared access signature.');
+        }
+        safeCallback(_callback, err, result);
+      });
+    }, callback);
+  }
+
+  /**
+   * @description      The `notifyBlobUploadStatus` method sends IoT Hub the result of a blob upload.
+   * @param {boolean}                        isSuccess          The success or failure status from the storage blob operation result.
+   * @param {number}                         statusCode         The HTTP status code associated with the storage blob result.
+   * @param {string}                         statusDescription  The description of the HTTP status code.
+   * @param {ErrorCallback}                  [callback]         Optional callback to call when the upload is complete.
+   * @returns {Promise<void> | void}                            Promise if no callback function was passed, void otherwise.
+   *
+   * @throws {ReferenceException} If uploadResponse is falsy.
+   */
+  notifyBlobUploadStatus(isSuccess: boolean, statusCode: number, statusDescription: string, callback: ErrorCallback): void;
+  notifyBlobUploadStatus(isSuccess: boolean, statusCode: number, statusDescription: string): Promise<void>;
+  notifyBlobUploadStatus(isSuccess: boolean, statusCode: number, statusDescription: string, callback?: ErrorCallback): Promise<void> | void {
+    return callbackToPromise((_callback) => {
+      /*Codes_SRS_NODE_DEVICE_CLIENT_41_005: [The `notifyBlobUploadStatus` method shall throw a `ReferenceError` if `isSuccess` is falsy but not the boolean false.]*/
+      /*Codes_SRS_NODE_DEVICE_CLIENT_41_006: [The `notifyBlobUploadStatus` method shall throw a `ReferenceError` if `statusCode` is falsy but not the number 0.]*/
+      /*Codes_SRS_NODE_DEVICE_CLIENT_41_007: [The `notifyBlobUploadStatus` method shall throw a `ReferenceError` if `statusDescription` is falsy but not an empty string.]*/
+      if (!isSuccess && typeof(isSuccess) !== 'boolean' ) throw new ReferenceError('isSuccess cannot be \' ' + isSuccess + ' \'');
+      if (!statusCode && !(statusCode === 0)) throw new ReferenceError('statusCode cannot be \' ' + statusCode + ' \'');
+      if (!statusDescription && statusDescription !== '') throw new ReferenceError('statusDescription cannot be \' ' + statusDescription + ' \'.');
+      /*Codes_SRS_NODE_DEVICE_CLIENT_41_012: [The `notifyBlobUploadStatus` method shall throw a `ReferenceError` if `correlationId` is not set.]*/
+      if (!this._blobStorageUploadParams || !this._blobStorageUploadParams.correlationId) {
```

Comment:
See above: I think this should be a parameter. The app may need to manage multiple uploads: some larger (slower) and some smaller. It's ideal to allow the app to do these in parallel.